pyproject.toml

```diff
@@ -37,7 +37,7 @@ classifiers = [
 dependencies = [
     "numpy>=1.23.5,<2.0.0",
     "scipy>=1.11.4",
-    "casadi>=3.6.6",
+    "casadi>=3.6.7",
```

Comment:
Since we are linking to the C++ code, shouldn't this be a hard pin at 3.6.7? I looked at this before, but seems like a good time to do it

Reply:
Yes, it should be. The problem is that if CasADi 3.7.X is released and we don't update it later, the mismatch will break things at runtime across all platforms (because CasADi libs are not compatible across minor versions, as we noticed in #3100). So I don't recommend doing this right now, a better time would be when we switch the build system and move away from `vcpkg`. Not pinning like it is currently would break just Windows, and would allow macOS and Linux builds from source to work.

The proper solution is:
- either, pin CasADi for both build-time and runtime, for which `conda-forge` arguably offers a better method with `{{ pin_compatible }}`. I'd like to do that in a separate PR after carefully considering it because it would be a non-weak constraint for users.
- or, not use the CasADi Python package as a build-time dependency but build it as its own extension module (like we do for SuiteSparse and SUNDIALS).